lib/dom-to-react.js

```diff
@@ -51,15 +52,23 @@ function domToReact(nodes, options) {
     }
 
     if (node.type === 'text') {
-      // if trim option is enabled, skip whitespace text nodes
-      if (trim) {
-        data = node.data.trim();
-        if (data) {
-          result.push(node.data);
-        }
-      } else {
-        result.push(node.data);
+      isWhitespace = !node.data.trim().length;
+
+      if (isWhitespace && node.parent && !canTextBeChildOfNode(node.parent)) {
```

Comment:
Can we do the `node.parent` check in `canTextBeChildOfNode()`?

Reply:
I responded to another comment you had related to this. Let me know what you think.

I could change the implementation to accept an `undefined` node but that wouldn't be really correct since `canTextBeChildOfNode(undefined)` would have to return `true`. This is incorrect because you can't actually have a text node as a child of something that undefined.

Reply:
Ggoing to resolve this conversation and keep discussion in https://github.com/remarkablemark/html-react-parser/pull/394#discussion_r769201694